AElf.Sdk.CSharp.Tests/ContextTests.cs

```diff
@@ -0,0 +1,311 @@
+using System;
+using AElf.Common;
+using AElf.Cryptography;
+using AElf.Cryptography.ECDSA;
+using AElf.Kernel;
+using AElf.Kernel.Blockchain.Application;
+using AElf.Kernel.SmartContract;
+using AElf.Kernel.SmartContract.Application;
+using AElf.Kernel.Types;
+using AElf.Types.CSharp;
+using Google.Protobuf;
+using Shouldly;
+using Xunit;
+
+namespace AElf.Sdk.CSharp.Tests
+{
+    public class ContextTests : SdkCSharpTestBase
+    {
+        private readonly Context _context;
+        private readonly IBlockchainService _blockchainService;
+        private readonly ISmartContractService _smartContractService;
+        private int _chainId = 1;
+        private readonly ECKeyPair _keyPair;
+
+        public ContextTests()
+        {
+            _blockchainService = GetRequiredService<IBlockchainService>();
+            _smartContractService = GetRequiredService<ISmartContractService>();
+            _keyPair = CryptoHelpers.GenerateKeyPair();
+        }
+
+        [Fact]
+        public void Should_Transaction_ReadOnly()
+        {
+            var context = CreateNewContext();
+            var transaction = context.Transaction;
+            transaction.From = Address.FromString("NewFrom");
+            
+            context.Transaction.ShouldNotBe(transaction);
+        }
+        
+        [Fact]
+        public void Should_TransactionFrom_ReadOnly()
+        {
+            var context = CreateNewContext();
+            var from = context.Transaction.From;
+            from = Address.FromString("NewFrom");
+            
+            context.Transaction.From.ShouldNotBe(from);
+        }
+        
+        [Fact]
+        public void Should_ContractAddress_ReadOnly()
+        {
+            var context = CreateNewContext();
+            var contractAddress = context.SmartContractContext.ContractAddress;
+            contractAddress = Address.FromString("NewContractAddress");
+            
+            context.SmartContractContext.ContractAddress.ShouldNotBe(contractAddress);
+        }
+        
+        [Fact]
+        public void Should_PreviousBlockHash_ReadOnly()
+        {
+            var context = CreateNewContext();
+            var previousBlockHash = context.PreviousBlockHash;
```

Comment:
```
 var previousBlockHash = context.PreviousBlockHash;
            previousBlockHash = Hash.FromString("NewHash");
```

this lines can not test readonly property.

readonly property and writable property will work the same as your code

PLS make more attention on the logic, but not a property is readonly or not

Reply:
It has been removed. Maybe My test is not correct, but I think test read only is necessary, If someone deleted it by mistake, It is dangerous.